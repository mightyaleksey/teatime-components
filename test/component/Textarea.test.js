'use strict';

const Textarea = require('../../component/Textarea');
const React = require('react');
const renderer = require('react-test-renderer');

test('Textarea renders correctly with defaults', () => {
  const tree = renderer.create(
    <Textarea
      name='textarea control'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Textarea renders correctly with defined props', () => {
  const tree = renderer.create(
    <Textarea
      autoComplete='off'
      autoFocus={true}
      className='outer'
      cols={5}
      disabled={true}
      id='textareas id'
      maxLength={25}
      name='textarea control'
      placeholder='it is a placeholder'
      readOnly={false}
      rows={10}
      value='custom value'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Textarea\'s initial size is `m`', () => {
  expect(Textarea.defaultProps.size).toBe('m');
});
