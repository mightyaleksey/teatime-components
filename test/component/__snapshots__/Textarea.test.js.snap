exports[`test Textarea renders correctly with defaults 1`] = `
<textarea
  autoComplete={undefined}
  autoFocus={undefined}
  className="control"
  cols={undefined}
  defaultValue={undefined}
  disabled={undefined}
  id={undefined}
  maxLength={undefined}
  name="textarea control"
  onChange={[Function]}
  placeholder={undefined}
  readOnly={undefined}
  rows="3"
  value={undefined} />
`;

exports[`test Textarea renders correctly with defined props 1`] = `
<textarea
  autoComplete="off"
  autoFocus={true}
  className="control outer"
  cols={5}
  defaultValue={undefined}
  disabled={true}
  id="textareas id"
  maxLength={25}
  name="textarea control"
  onChange={[Function]}
  placeholder="it is a placeholder"
  readOnly={false}
  rows={10}
  value="custom value" />
`;
